sbpy/data/core.py

```diff
@@ -614,17 +614,20 @@ def __repr__(self):
 
     def __getitem__(self, ident):
         """Return columns or rows from data table(``self._table``); checks
-        for and may use alternative field names."""
+        for and may use alternative field names. This method will always return
+        an instance of __class__, except in the case when a field name is
+        requested (then return a Table column)."""
 
+        # ignore slices
+        if isinstance(ident, slice):
+            pass
```

Comment:
Looks good.  One small thing, why are slice and int checked in the conditional statement if there is a pass and the new instance `self.from_table(self._table[ident])` will be returned?

Reply:
Those `if` statements were just for me to check on the behavior for different cases. As of bcc8538 this is condensed in a more useful structure.